docs/adr/template/adr-template.md

```diff
@@ -0,0 +1,75 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: "{proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)}"
```

Comment:
I wonder how useful some of these statuses are? Would a better set be something like:
```
status: "{proposed | implemented | superseded by [ADR-0005](0005-example.md)}"
```

Where "proposed" means "we've decided to do this, but not built it yet", and "implemented" means "the current version of the API includes this change"?

Reply:
Indeed - "accepted | rejected" are catered for and documented by PR review audit trail.  May be useful to keep "deprecated" as an option?  Although it's likely that deprecation will involve superseding with a new ADR to explain the decision-making process.

Reply:
"rejected" seems an important status for public domain discussion. We may find certain design decisions are questioned repeatedly, and having a "rejected" ADR to point at is better than producing a justification every time.

I'm not sure "implemented" is needed, as an "accepted" ADR can be updated with implementation details.

Your definition of "proposed" here implies that all the discussion takes place elsewhere, and only an accepted proposal makes it to an ADR. Line 11 in `Contributing.md` indicates that an ADR might be the start of a process which results in a change being included, or not. These two changes are not compatible.

Reply:
Alternative view: Is it worth keeping rejected. It can be as important to record the reasoning for not doing something as for doing something.

Reply:
**On "rejected"**
I'm inclined to agree with @AndrewGibb and @j616 here, where the subtlety is that a "rejected" ADR only gets merged if it's useful to retain the record. To use an example in the limit, a vexatious PR to make a stylistic change would just get closed, a seriously considered change to the model that we decide against would become "rejected"

**On "proposed"/"accepted"** 
> Your definition of "proposed" here implies that all the discussion takes place elsewhere, and only an accepted proposal makes it to an ADR. Line 11 in Contributing.md indicates that an ADR might be the start of a process which results in a change being included, or not. These two changes are not compatible.

I think the distinction is that once you've completed that process in Contributing.md, you have a PR for discussion, but the process of including or not including the change doesn't end until the document appears on `main`

Does that make sense?

**On "implemented"**
> I'm not sure "implemented" is needed, as an "accepted" ADR can be updated with implementation details.

I suppose the presence or absence of a PR link in the Implementation section serves that purpose, making it redundant.

Reply:
> I think the distinction is that once you've completed that process in Contributing.md, you have a PR for discussion, but the process of including or not including the change doesn't end until the document appears on main

> Does that make sense?

That leads to the following set of conditions:

<table>
<tr>
 <td> PR status
 <td> adr status
 <td> actual meaning
<tr>
 <td> Not closed
 <td> "proposed"
 <td> A proposal, for discussion
<tr>
 <td> Closed
 <td> "proposed"
 <td> Proposal accepted
<tr>
 <td> Closed
 <td> "rejected"
 <td> Proposal rejected
</table>

Which doesn't make much sense to me. Including "accepted" for accepted proposals makes the meaning of the status much more clear.

Reply:
Ok, I think the conclusion is we keep these as they are for now. Thanks for your insight everyone!